### ISSUE TITLE
error upgrade to 2.1.25.2

### ISSUE THREAD

**Issue by p0larf0x** [nilsteampassnet/TeamPass#1236]
**Hello! When I upgrade TP from 2.1.23 to 2.1.25.2, I get error:**
Misc table will be populated with new values - An error appears when inserting datas! 

**OS:**
Centos 6.4

**packages:**
nginx-1.4.4-1.el6.ngx.x86_64
php-5.4.38-1.el6.remi.x86_64

**Log nginx error:**
2016/03/13 22:21:43 [error] 4643#0: *636466 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught exception 'configFileNotFoundException' with message 'configuration file not found for CSRFProtector!' in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php:82
Stack trace:
#0 /data/www/html/teampass/index.php(31): csrfProtector::init()
#1 {main}

  thrown in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 82" while reading response header from upstream, client: 10.1.1.240, server: teampass, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9028", host: "teampass"
2016/03/13 22:37:36 [error] 4627#0: *637190 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught exception 'configFileNotFoundException' with message 'configuration file not found for CSRFProtector!' in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php:82
Stack trace:
#0 /data/www/html/teampass/index.php(31): csrfProtector::init()
#1 {main}

  thrown in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 82" while reading response header from upstream, client: 10.1.1.240, server: teampass, request: "GET /index.php?session=expired HTTP/1.1", upstream: "fastcgi://127.0.0.1:9028", host: "teampass", referrer: "http://teampass/index.php?page=items"
2016/03/13 22:59:41 [error] 4643#0: *638430 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught exception 'configFileNotFoundException' with message 'configuration file not found for CSRFProtector!' in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php:82
Stack trace:
#0 /data/www/html/teampass/index.php(31): csrfProtector::init()
#1 {main}

  thrown in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 82" while reading response header from upstream, client: 10.1.1.240, server: teampass, request: "GET /index.php?session=expired HTTP/1.1", upstream: "fastcgi://127.0.0.1:9028", host: "teampass", referrer: "http://teampass/index.php?page=items"
2016/03/13 22:59:44 [error] 4643#0: *638430 open() "/data/www/html/teampass/favicon.ico" failed (2: No such file or directory), client: 10.1.1.240, server: teampass, request: "GET /favicon.ico HTTP/1.1", host: "teampass"
2016/03/13 22:59:49 [error] 4643#0: *638430 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught exception 'configFileNotFoundException' with message 'configuration file not found for CSRFProtector!' in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php:82
Stack trace:
#0 /data/www/html/teampass/index.php(31): csrfProtector::init()
#1 {main}

  thrown in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 82" while reading response header from upstream, client: 10.1.1.240, server: teampass, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9028", host: "teampass"
2016/03/13 23:00:01 [error] 4643#0: *638430 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught exception 'configFileNotFoundException' with message 'configuration file not found for CSRFProtector!' in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php:82
Stack trace:
#0 /data/www/html/teampass/index.php(31): csrfProtector::init()
#1 {main}

  thrown in /data/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 82" while reading response header from upstream, client: 10.1.1.240, server: teampass, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9028", host: "teampass"

**Comment by OlegSmirnov** [comment]
System clearly doesn't have access to files. I would check
1) pathes
2) file permissions
3) SELinux context

**Comment by p0larf0x** [comment]
Thank you, tonight check!

**Comment by p0larf0x** [comment]
launched a test stand similar production server, uploaded to the database and configuration of include.
1) pathes default
2) Rights already put 777
3) SELinux disabled
The error is the same:
Misc table will be populated with new values - An error appears when inserting datas!
But in the log anything anymore

**Comment by p0larf0x** [comment]
On apache test stand update sucsesfull. Truble in my php-fpm or nginx

**Comment by zhanko73** [comment]
The csrfp.config.php is missing, and there is a sample instead called csrfp.config.sample.php 
If the csrfp.config.php is missing then the sample is supposed to be coped as csrfp.config.php and fill some part like ssl token. Unfortunately it seems the length of the ssl token is not updated so the 25 byte long token is converted to hexa where the length will be 50 characters. This will cause additional problem.
The possible root cause of the php file missing: the directory that supposed to store that file did not have enought rights.
I issued this actually, but never tested with install script after, rather I fixed them manually.
chmod 777 install/ includes/ includes/avatars/ files/ upload/ 

Zoli
